plonk-core/src/proof_system/pi.rs

```diff
@@ -114,6 +114,34 @@ where
         let evals = self.as_evals(n);
         DensePolynomial::from_coefficients_vec(domain.ifft(&evals))
     }
+
+    /// Constructs [`PublicInputs`] from a positions and a values.
+    ///
+    /// Panics if the positions and values have different lenghts or if
+    /// several values try to be inserted in the same position.
+    pub fn from_val_pos<T>(pos: &[usize], vals: &[T]) -> Result<Self, Error>
```

Comment:
Not sure this is better, but still
```suggestion
    /// Constructs [`PublicInputs`] from positions and values.
    ///
    /// Panics if the positions and values have different lenghts or if
    /// `pos` contains duplicates (which by the way should not be
    /// possible when using positions returned by circuit compilation). 
    pub fn from_val_pos<T>(pos: &[usize], vals: &[T]) -> Result<Self, Error>
```

Reply:
The panic resulting from several values being inserted in the same position not only occurs if there are repeated elements in `pos`. This is a consequence of having `vals` accept elements that implement `ToConstraintField` instead of field elements directly. A single values may occupy several positions.
We can for example to insert directly a point of the inner curve and it will *start* the given position but will occupy 2, one for each coordinates. If we then try to insert some value at the next position we get an error because it is already occupied by the 2nd coordinate of the point.
Maybe this behavior is not clear and should be more clearly stated. WDYT @iamnotacake ?

Reply:
Didn't notice at first that `T` is not `F`. Well, unless user misuses this method (i.e. puts ECC point instead of `F`) it should work fine. But then, another question arises: how would this same user create `&[T]` if `T` could have different size for different types? Maybe example of this in function doc would make it a bit better

Reply:
A user could put `ECC` points and it's not a misuse of the method. It is true that one cannot combine different types (ecc points and field elements) and use this method to create the PI in one call. In that case everything should be converted first to `F`.
This function can be used just as if it received `&[F]` but it also supports cases where the PI are just some other type.

Reply:
> It is true that one cannot combine different types (ecc points and field elements) and use this method to create the PI in one call

Then we'll think about that in future, if combining both types and converting points to `F`s will be not that easy, with non-obvious mistakes and so on.